### PR TITLE
Added com.amazon.aws.lambda/java-context/jsonschema/1-0-0 (closes #309)

### LINKED ISSUE #309: Schema for a com.amazon.aws.lambda/java_context/jsonschema/1-0-0
Schema for an [AWS lambda context](http://docs.aws.amazon.com/lambda/latest/dg/java-context-object.html)

### PR DESCRIPTION
**Pull request by ninjabear**

### CODE REVIEW

File: schemas/com.amazon.aws.lambda/java-context/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "$schema" : "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "self" : {
+    "vendor" : "com.amazon.aws.lambda",
+    "name" : "java-context",
```
Comment (alexanderdean) [inline schemas/com.amazon.aws.lambda/java-context/jsonschema/1-0-0]: I think we should use `_` for consistency with the other `com.amazon.aws.*` schema(s)

File: schemas/com.amazon.aws.lambda/java-context/jsonschema/1-0-0 [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "$schema" : "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
```
Comment (alexanderdean) [inline schemas/com.amazon.aws.lambda/java-context/jsonschema/1-0-0]: Can you add a description line with link back to the Lambda documentation? See e.g. https://github.com/snowplow/iglu-central/blob/master/schemas/com.segment/alias/jsonschema/1-0-0#L3

File: schemas/com.amazon.aws.lambda/java-context/jsonschema/1-0-0 [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    "remainingTimeMillis" : {
+      "type" : "integer",
+      "maximum" : 2147483647,
+      "minimum" : 0
+    },
+    "logGroupName" : {
+      "type" : "string"
+    },
+    "invokedTimeUTC" : {
```
Comment (alexanderdean) [inline schemas/com.amazon.aws.lambda/java-context/jsonschema/1-0-0]: I think normally we would say `invokedTimeUtc` (like we say `Url` not `URL`)

File: schemas/com.amazon.aws.lambda/java-context/jsonschema/1-0-0 [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      "type" : "string",
+      "format" : "date-time"
+    },
+    "memoryLimitInMB" : {
```
Comment (alexanderdean) [inline schemas/com.amazon.aws.lambda/java-context/jsonschema/1-0-0]: I think normally we would say `memoryLimitInMb` (like we say `Url` not `URL`)

**Comment by ninjabear** [comment]
Hey @alexanderdean -

* I've added a description field 
   * https://github.com/snowplow/iglu-central/commit/b7a4f360d0dd610390723008d99e78b8ea224527#diff-a377ddb0122a0dd49daf7e76003f8da5R3
* Kept the fields named as closely to the getters as possible
* Also included the optional fields
   * client context - this contains a client object and two hashmaps (environment and custom) which can contain anything. I don't think we can schema these
   * identity - two strings
* tabified the json inkeeping with other schemas

File: schemas/com.amazon.aws.lambda/java_context/jsonschema/1-0-0 [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+							"type": "string"
+						},
+						"appPackageName": {
+							"type": "string"
+						}
+					},
+					"additionalProperties": false
+				}
+			},
+			"additionalProperties": true
```
Comment (alexanderdean) [inline schemas/com.amazon.aws.lambda/java_context/jsonschema/1-0-0]: Why is this one `"additionalProperties": true` but the rest `: false`?

**Comment by alexanderdean** [comment]
`two hashmaps (environment and custom) which can contain anything` - do you mean they are `Map<String, Object>`? That is expressible in JSON Schema - ping @chuwy for details!

**Comment by ninjabear** [comment]
They're Map<String,String>, so environment is the current environment variables, and custom is a key-value set of custom values

**Comment by alexanderdean** [comment]
Cool - that's very easy to represent in JSON Schema...

**Comment by ninjabear** [comment]
Hey @alexanderdean - I've added [the key maps](https://github.com/snowplow/iglu-central/blob/feature/aws-lambda-context/schemas/com.amazon.aws.lambda/java_context/jsonschema/1-0-0#L53-L68). I think this is ready to merge now - unless you can see anything wrong with it?

**Comment by alexanderdean** [comment]
Hi @ninjabear - awesome, I will do a release!

**Comment by alexanderdean** [comment]
Hey @ninjabear - I need the SQL and JSON Paths too to avoid breaking the load into Redshift...

**Comment by alexanderdean** [comment]
Hey @ninjabear - I've added the tickets here:

- https://github.com/snowplow/snowplow/issues/2559
- https://github.com/snowplow/snowplow/issues/2560

I ran Anton's latest Schema Guru over the JSON to produce them...

**Pull request merged**